app/validators/allowed_email_domain_validator.rb

```diff
@@ -0,0 +1,16 @@
+class AllowedEmailDomainValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    if value.present?
+      return if value =~ /\.gov\.uk\z/i
+
+      # TODO: we might not want to check against current user domain
+      # when we have a proper allow list?
+      if record.respond_to?(:current_user) && record.current_user.present?
+        user_domain = record.current_user.email.then { |email| email[email.rindex("@")..] }
+        return if user_domain && value.end_with?(user_domain)
```

Comment:
Will this be vulnerable to malicious domains that end with the user's domain? i.e. if the user is `david@agovernmentwebsite.co.uk` could the submission email be`hacker@notagovernmentwebsite.co.uk`?

Reply:
I don't _think_ so, I tried to check that with the test at https://github.com/alphagov/forms-admin/blob/9f93faa39294ee76f9c6bd123891a61ea7c2324f/spec/validator/allowed_email_domain_validator_spec.rb#L49-L52

Reply:
I've tweaked the code here slightly to make it clearer that the string should include the `@` symbol

Reply:
Sorry, didn't spot that test! That change is helpful though so thanks for updating